.github/workflows/build.yml

```diff
@@ -26,5 +26,26 @@ jobs:
         if: failure()
         uses: actions/upload-artifact@v4
         with:
-          name: rat
-          path: target/rat.txt
\ No newline at end of file
+          name: rat-${{ matrix.java }}
+          path: target/rat.txt
+
+      - name: Upload Rat Report On Failure
+        if: failure()
+        uses: actions/upload-artifact@v4
+        with:
+          name: rat-distribution-${{ matrix.java }}
+          path: artemis-console-distribution/target/rat.txt
+
+      - name: Upload Rat Report On Failure
+        if: failure()
+        uses: actions/upload-artifact@v4
+        with:
+          name: rat-extension-${{ matrix.java }}
+          path: artemis-console-extension/target/rat.txt
+
+      - name: Upload Rat Report On Failure
+        if: failure()
+        uses: actions/upload-artifact@v4
+        with:
+          name: rat-console-${{ matrix.java }}
+          path: artemis-console-war/target/rat.txt
```

Comment:
Having lots of steps doing essentially the same overall thing, and which are likely to individually fail themselves if the file doenst exist (job could fail for other reasons) seems less than ideal. The per-module nature of it is also brittle in terms of handling any future additions and renames.

Should be able to do it in one step, with one upload, and which doesnt fail if there are no rat.txt files.

https://github.com/actions/upload-artifact?tab=readme-ov-file#upload-using-a-wildcard-pattern
https://github.com/actions/upload-artifact?tab=readme-ov-file#customization-if-no-files-are-found

Reply:
I will fix this monday

Reply:
fixed